compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/transform/GBKTransform.java

```diff
@@ -235,6 +234,7 @@ private void triggerTimers(final Instant processingTime,
                             final Instant synchronizedTime,
                             final Watermark watermark) {
 
+    ArrayList<K> removed = new ArrayList<>();
```

Comment:
Instead of creating this, we can remove the element in a while loop with `Iterator.remove` : https://stackoverflow.com/questions/23665098/remove-an-element-from-a-list-inside-a-loop/23665370